tests/scene/test_sky.h

```diff
@@ -130,10 +131,84 @@ TEST_CASE("[SceneTree][Sky] Material setter and getter") {
 
 		CHECK(test_sky->get_material() == material2);
 		CHECK(sky2->get_material() == material1);
-		memdelete(sky2);
 	}
+}
+
+TEST_CASE("[SceneTree][Sky] Invalid radiance size handling") {
+	Ref<Sky> test_sky;
+	test_sky.instantiate();
+
+	// Attempt to set an invalid radiance size.
+	ERR_PRINT_OFF;
+	test_sky->set_radiance_size(Sky::RADIANCE_SIZE_MAX);
+	ERR_PRINT_ON;
+
+	// Verify that the radiance size remains unchanged.
+	CHECK(test_sky->get_radiance_size() == Sky::RADIANCE_SIZE_256);
+}
+
+TEST_CASE("[SceneTree][Sky] Process mode variations") {
+	Ref<Sky> test_sky;
+	test_sky.instantiate();
+
+	// Test all process modes.
+	const Sky::ProcessMode process_modes[] = {
+		Sky::PROCESS_MODE_AUTOMATIC,
+		Sky::PROCESS_MODE_QUALITY,
+		Sky::PROCESS_MODE_INCREMENTAL,
+		Sky::PROCESS_MODE_REALTIME
+	};
+
+	for (Sky::ProcessMode mode : process_modes) {
+		test_sky->set_process_mode(mode);
+		CHECK(test_sky->get_process_mode() == mode);
+	}
+}
+
+TEST_CASE("[SceneTree][Sky] Radiance size variations") {
+	Ref<Sky> test_sky;
+	test_sky.instantiate();
+
+	// Test all radiance sizes except MAX.
+	const Sky::RadianceSize radiance_sizes[] = {
+		Sky::RADIANCE_SIZE_32,
+		Sky::RADIANCE_SIZE_64,
+		Sky::RADIANCE_SIZE_128,
+		Sky::RADIANCE_SIZE_256,
+		Sky::RADIANCE_SIZE_512,
+		Sky::RADIANCE_SIZE_1024,
+		Sky::RADIANCE_SIZE_2048
+	};
+
+	for (Sky::RadianceSize size : radiance_sizes) {
+		test_sky->set_radiance_size(size);
+		CHECK(test_sky->get_radiance_size() == size);
+	}
+}
+
+TEST_CASE("[SceneTree][Sky] Null material handling") {
+	Ref<Sky> test_sky;
+	test_sky.instantiate();
+
+	SUBCASE("Setting null material") {
+		test_sky->set_material(Ref<Material>());
+		CHECK(test_sky->get_material().is_null());
+	}
+
+	SUBCASE("Overwriting existing material with null") {
+		Ref<Material> material = memnew(Material);
```

Comment:
```suggestion
		Ref<Material> material;
		material.instantiate();
```

Reply:
Thanks for the suggestion! Removed all `memnew()` and `memdelete()` now.